File: go/problems/day6_test.go
package problems

import "testing"

var day6 Solvable[int] = Day6{}

func TestDay6_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day6_1")
	result := day6.Solve(input)

	var expected int = 288
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay6_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day6_2")
	result := day6.Solve(input)

	var expected int = 138915
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day6.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day6.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 6", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day6_1");
    const result = solve(input);
    assert.equal(result, 288);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day6_2");
    const result = solve(input);
    assert.equal(result, 138915);
  });
});
